File: server/internal/interface/admin/http/router/v1/content.go
package v1

import (
	"github.com/gin-gonic/gin"

	"netyadmin/internal/interface/admin/http/handler/v1/content"
)

type ContentRouter struct {
	category    *content.ContentCategoryHandler
	article     *content.ContentArticleHandler
	bannerGroup *content.ContentBannerGroupHandler
	bannerItem  *content.ContentBannerItemHandler
}

func NewContentRouter(
	category *content.ContentCategoryHandler,
	article *content.ContentArticleHandler,
	bannerGroup *content.ContentBannerGroupHandler,
	bannerItem *content.ContentBannerItemHandler,
) *ContentRouter {
	return &ContentRouter{
		category:    category,
		article:     article,
		bannerGroup: bannerGroup,
		bannerItem:  bannerItem,
	}
}

func (r *ContentRouter) RegisterPublic(group *gin.RouterGroup) {}

func (r *ContentRouter) RegisterAuth(group *gin.RouterGroup) {}

func (r *ContentRouter) RegisterPermission(group *gin.RouterGroup) {
	r.registerCategory(group)
	r.registerArticle(group)
	r.registerBannerGroup(group)
	r.registerBannerItem(group)
}

func (r *ContentRouter) registerCategory(group *gin.RouterGroup) {
	categoryGroup := group.Group("/content/categories")
	{
		categoryGroup.GET("", r.category.List)
		categoryGroup.GET("/tree", r.category.Tree)
		categoryGroup.GET("/:id", r.category.Get)
		categoryGroup.POST("", r.category.Create)
		categoryGroup.PUT("/:id", r.category.Update)
		categoryGroup.DELETE("/:id", r.category.Delete)
	}
}

func (r *ContentRouter) registerArticle(group *gin.RouterGroup) {
	articleGroup := group.Group("/content/articles")
	{
		articleGroup.GET("", r.article.List)
		articleGroup.GET("/:id", r.article.Get)
		articleGroup.POST("", r.article.Create)
		articleGroup.PUT("/:id", r.article.Update)
		articleGroup.DELETE("/:id", r.article.Delete)
		articleGroup.PUT("/:id/publish", r.article.Publish)
		articleGroup.PUT("/:id/unpublish", r.article.Unpublish)
		articleGroup.PUT("/:id/top", r.article.SetTop)
	}
}

func (r *ContentRouter) registerBannerGroup(group *gin.RouterGroup) {
	bannerGroupGroup := group.Group("/content/banner-groups")
	{
		bannerGroupGroup.GET("", r.bannerGroup.List)
		bannerGroupGroup.GET("/:id", r.bannerGroup.Get)
		bannerGroupGroup.POST("", r.bannerGroup.Create)
		bannerGroupGroup.PUT("/:id", r.bannerGroup.Update)
		bannerGroupGroup.DELETE("/:id", r.bannerGroup.Delete)
	}
}

func (r *ContentRouter) registerBannerItem(group *gin.RouterGroup) {
	bannerItemGroup := group.Group("/content/banner-items")
	{
		bannerItemGroup.GET("", r.bannerItem.List)
		bannerItemGroup.GET("/:id", r.bannerItem.Get)
		bannerItemGroup.POST("", r.bannerItem.Create)
		bannerItemGroup.PUT("/:id", r.bannerItem.Update)
		bannerItemGroup.DELETE("/:id", r.bannerItem.Delete)
	}
}

File: admin-web/src/service/api/v1/content.ts
import type { Content } from '@/typings/api/v1/content';
import { request } from '../../request';

/**
 * Content Article
 */

export function fetchGetArticleList(params?: Content.ArticleSearchParams) {
  return request<Content.ArticleList>({
    url: '/admin/v1/content/articles',
    method: 'get',
    params
  });
}

export function fetchGetArticle(id: number) {
  return request<Content.Article>({
    url: `/admin/v1/content/articles/${id}`,
    method: 'get'
  });
}

export function fetchCreateArticle(data: Content.CreateArticleParams) {
  return request<Content.Article>({
    url: '/admin/v1/content/articles',
    method: 'post',
    data
  });
}

export function fetchUpdateArticle(id: number, data: Content.UpdateArticleParams) {
  return request<Content.Article>({
    url: `/admin/v1/content/articles/${id}`,
    method: 'put',
    data
  });
}

export function fetchDeleteArticle(id: number) {
  return request({
    url: `/admin/v1/content/articles/${id}`,
    method: 'delete'
  });
}

export function fetchPublishArticle(id: number) {
  return request({
    url: `/admin/v1/content/articles/${id}/publish`,
    method: 'put'
  });
}

export function fetchUnpublishArticle(id: number) {
  return request({
    url: `/admin/v1/content/articles/${id}/unpublish`,
    method: 'put'
  });
}

export function fetchSetArticleTop(id: number, data: Content.SetArticleTopParams) {
  return request({
    url: `/admin/v1/content/articles/${id}/top`,
    method: 'put',
    data
  });
}

/**
 * Content Banner Group
 */

export function fetchGetBannerGroupList(params?: Content.BannerGroupSearchParams) {
  return request<Content.BannerGroupList>({
    url: '/admin/v1/content/banner-groups',
    method: 'get',
    params
  });
}

export function fetchGetBannerGroup(id: number) {
  return request<Content.BannerGroup>({
    url: `/admin/v1/content/banner-groups/${id}`,
    method: 'get'
  });
}

export function fetchGetAllBannerGroups() {
  return request<Content.BannerGroup[]>({
    url: '/admin/v1/content/banner-groups',
    method: 'get',
    params: { size: 100 }
  });
}

export function fetchCreateBannerGroup(data: Content.CreateBannerGroupParams) {
  return request<Content.BannerGroup>({
    url: '/admin/v1/content/banner-groups',
    method: 'post',
    data
  });
}

export function fetchUpdateBannerGroup(id: number, data: Content.UpdateBannerGroupParams) {
  return request<Content.BannerGroup>({
    url: `/admin/v1/content/banner-groups/${id}`,
    method: 'put',
    data
  });
}

export function fetchDeleteBannerGroup(id: number) {
  return request({
    url: `/admin/v1/content/banner-groups/${id}`,
    method: 'delete'
  });
}

/**
 * Content Banner Item
 */

export function fetchGetBannerItemList(params?: Content.BannerItemSearchParams) {
  return request<Content.BannerItemList>({
    url: '/admin/v1/content/banner-items',
    method: 'get',
    params
  });
}

export function fetchGetBannerItem(id: number) {
  return request<Content.BannerItem>({
    url: `/admin/v1/content/banner-items/${id}`,
    method: 'get'
  });
}

export function fetchCreateBannerItem(data: Content.CreateBannerItemParams) {
  return request<Content.BannerItem>({
    url: '/admin/v1/content/banner-items',
    method: 'post',
    data
  });
}

export function fetchUpdateBannerItem(id: number, data: Content.UpdateBannerItemParams) {
  return request<Content.BannerItem>({
    url: `/admin/v1/content/banner-items/${id}`,
    method: 'put',
    data
  });
}

export function fetchDeleteBannerItem(id: number) {
  return request({
    url: `/admin/v1/content/banner-items/${id}`,
    method: 'delete'
  });
}

/**
 * Content Category
 */

export function fetchGetCategoryList(params?: Content.CategorySearchParams) {
  return request<Content.CategoryList>({
    url: '/admin/v1/content/categories',
    method: 'get',
    params
  });
}

export function fetchGetCategoryTree(refresh = false) {
  return request<Content.CategoryTree[]>({
    url: '/admin/v1/content/categories/tree',
    method: 'get',
    params: { refresh }
  });
}

export function fetchGetCategory(id: number) {
  return request<Content.Category>({
    url: `/admin/v1/content/categories/${id}`,
    method: 'get'
  });
}

export function fetchCreateCategory(data: Content.CreateCategoryParams) {
  return request<Content.Category>({
    url: '/admin/v1/content/categories',
    method: 'post',
    data
  });
}

export function fetchUpdateCategory(id: number, data: Content.UpdateCategoryParams) {
  return request<Content.Category>({
    url: `/admin/v1/content/categories/${id}`,
    method: 'put',
    data
  });
}

export function fetchDeleteCategory(id: number) {
  return request({
    url: `/admin/v1/content/categories/${id}`,
    method: 'delete'
  });
}
